### PR TITLE
Vlanifns and dhcpns

### PR DESCRIPTION
**Pull request by oscar-lenin**
Added vlanifns adn dhcpns primitives

### CODE REVIEW

File: cloudcix_primitives/vlanifns.py [hunk 1]
```diff
@@ -0,0 +1,245 @@
...
+        fmt = PodnetErrorFormatter(
+            config_file,
+            podnet_node,
+            podnet_node == enabled,
+            {'payload_message': 'STDOUT', 'payload_error': 'STDERR'},
+            successful_payloads
+        )
+
+        payloads = {
+            'vlanif_check' : f'ip netns exec {namespace} ip link show {ifname}.{vlan}',
```
Comment (maria-walsh) [inline cloudcix_primitives/vlanifns.py:97]: @oscar-lenin  If the VLAN interface is linked the namespace, does this mean that is up?

Is it possible for it to be linked and down? If yes, to ensure it is up, should the `vlanif_up` payload always be executed?

For example, in the [ns primitive](https://github.com/CloudCIX/primitives/blob/d60c73c4c38b1341cfe2024ff30e1fbb09e77fcc/cloudcix_primitives/ns.py#L127), the `create_namespace` is only executed if the payload for `find_namespace` is unsuccessful. However, the remaining payloads are always executed to ensure it is in the expected state at the end of the build request.

File: cloudcix_primitives/vlanifns.py [hunk 2]
```diff
@@ -0,0 +1,245 @@
...
+            successful_payloads
+        )
+
+        payloads = {
+                'vlanif_check': f'ip netns exec {namespace} ip link show {ifname}.{vlan}',
+                'vlanif_del':  f'ip netns exec {namespace} ip link del {ifname}.{vlan}'
+        }
+
+
+        ret = rcc.run(payloads['vlanif_check'])
```
Comment (maria-walsh) [inline cloudcix_primitives/vlanifns.py:220]: https://github.com/CloudCIX/primitives/blob/d60c73c4c38b1341cfe2024ff30e1fbb09e77fcc/cloudcix_primitives/ns.py#L305

I think this pattern in the `ns` primitive for determining if a payload is required to be executed will be the one we will follow.  It reduces the need for exiting early with True, if other payloads are later required.

File: tools/test_vlanifns_primitive.py [hunk 3]
```diff
@@ -0,0 +1,48 @@
...
+
+cmd = sys.argv[1]
+
+namespace = 'mynetns'
+
+vlan = 4000
+ifname = 'private0'
+
+
+if len(sys.argv) > 2:
```
Comment (maria-walsh) [inline tools/test_vlanifns_primitive.py:22]:
```suggestion
if len(sys.argv) > 2:
    namespace = sys.argv[2]
if len(sys.argv) > 3:
    vlan = sys.argv[3]
if len(sys.argv) > 4:
    ifname = sys.argv[4]
```

Should it be extended to allow `vlan` and `ifname` to be arguments in the test request? 

e.g `tools/test_ns_primitive.py build mynetns 4000 private0`

File: cloudcix_primitives/vlanifns.py (hunk 1 above)
Comment (oscar-lenin) [reply cloudcix_primitives/vlanifns.py:97]: Hi Maria, that makes sense. I've changed it so vlanif_up is always run even if the interface already exists.

File: tools/test_vlanifns_primitive.py [hunk 4]
```diff
@@ -21,6 +21,11 @@
 
 if len(sys.argv) > 2:
     namespace = sys.argv[2]
+if len(sys.argv) > 3:
+    vlan = sys.argv[3]
+if len(sys.argv) > 4:
+    ifname = sys.argv[4]
+    namespace = sys.argv[2]
```
Comment (maria-walsh) [inline tools/test_vlanifns_primitive.py:28]:
```suggestion
```

I accidentally didn't remove this in my previous suggestion.

**Review by maria-walsh — APPROVED** [review]

**Pull request merged**